addons/calendar/models/res_users.py

```diff
@@ -46,15 +47,17 @@ def get_selected_calendars_partner_ids(self, include_user=True):
             partner_ids += [self.env.user.partner_id.id]
         return partner_ids
 
+    @api.model
+    def _default_user_calendar_default_privacy(self):
+        """ Get the calendar default privacy from the Default User Template, set public as default. """
+        if default_user := self.env.ref('base.default_user', raise_if_not_found=False):
+            return default_user.sudo().calendar_default_privacy or 'public'
+        return 'public'
+
     @api.model_create_multi
     def create(self, vals_list):
         """ Set the calendar default privacy as the same as Default User Template when defined. """
-        # Get the calendar default privacy from the Default User Template, set public as default.
-        default_privacy = 'public'
-        default_user = self.env.ref('base.default_user', raise_if_not_found=False)
-        if default_user and default_user.calendar_default_privacy:
-            default_privacy = default_user.calendar_default_privacy
-
+        default_privacy = self._default_user_calendar_default_privacy()
         # Update the dictionaries in vals_list with the calendar default privacy.
         for vals_dict in vals_list:
```

Comment:
I find it strange to have a default and a compute. I know we are in stable and avoid too big refactoring but the idea would be to have the compute computing the values (maybe with a precompute attribute)

Reply:
I think Flo used this idea for being able to re-use the function in two different scopes, `write` and in the `compute`. It would be better with precompute if we didn't have to use write indeed (that's the blocking point)

It is a very peculiar use case of checking the Default User Template's fields (and not the groups), it was first introduced in Odoo by myself as far as I know